LoRaEngine/LoraKeysManagerFacade/SearchDeviceByDevEUI.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoraKeysManagerFacade
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+    using LoRaWan.Shared;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Azure.WebJobs.Extensions.Http;
+    using Microsoft.Extensions.Logging;
+
+    public class SearchDeviceByDevEUI
+    {
+        [FunctionName(nameof(GetDeviceByDevEUI))]
+        public static async Task<IActionResult> GetDeviceByDevEUI([HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)]HttpRequest req, ILogger log, ExecutionContext context)
+        {
+            return await RunGetDeviceByDevEUI(req, log, context, ApiVersion.LatestVersion);
+        }
+
+        private static async Task<IActionResult> RunGetDeviceByDevEUI(HttpRequest req, ILogger log, ExecutionContext context, ApiVersion currentApiVersion)
+        {
+            try
+            {
+                VersionValidator.Validate(req);
+            }
+            catch (IncompatibleVersionException ex)
+            {
+                log.LogError(ex, "Invalid version");
+                return new BadRequestObjectResult(ex);
+            }
+
+            string devEUI = req.Query["DevEUI"];
+            if (string.IsNullOrEmpty(devEUI))
+            {
+                log.LogError("DevEUI missing in request");
+                return new BadRequestObjectResult("DevEUI missing in request");
+            }
+
+            var registryManager = LoRaRegistryManager.GetCurrentInstance(context.FunctionAppDirectory);
+
+            var result = new List<IoTHubDeviceInfo>();
+            var device = await registryManager.GetDeviceAsync(devEUI);
+            if (device != null)
+            {
+                if (device != null)
+                {
+                    result.Add(new IoTHubDeviceInfo()
+                    {
+                        DevEUI = devEUI,
+                        PrimaryKey = device.Authentication.SymmetricKey.PrimaryKey
+                    });
+                }
+
+                log.LogDebug($"Search for {devEUI} found 1 device");
+                return new OkObjectResult(result);
+            }
+            else
+            {
+                log.LogInformation($"Search for {devEUI} found 0 devices");
+                return new NotFoundObjectResult(result);
```

Comment:
As you mentioned, this will return 404 which means content not found. If clients are implemented correct that should be enough to indicate that there is no device with the requested DevEUI.

Reasoning behind returning an empty array as content is for those clients that are not checking the status code. They just serialise the return content to a list of devices that will be empty.